src/Tizen.NUI.Components/Controls/ItemsView/ItemsView.cs

```diff
@@ -0,0 +1,347 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Tizen.NUI.BaseComponents;
+using Tizen.NUI.Binding;
+
+namespace Tizen.NUI.Components
+{
+    /// <summary>
+    /// [Draft] This class provides a View that can layouting items in list and grid with high performance.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract class ItemsView : ScrollableBase, ICollectionChangedNotifier
+    {
+        /// <summary>
+        /// Base Constructor
+        /// </summary>
+       [EditorBrowsable(EditorBrowsableState.Never)]
+        public ItemsView() : base()
+        {
+            Scrolling += OnScrolling;
+        }
+
+        /// <summary>
+        /// Base Constructor with itemsSource
+        /// </summary>
+        /// <param name="itemsSource">item's data source</param>
+       [EditorBrowsable(EditorBrowsableState.Never)]
+        public ItemsView(IEnumerable itemsSource) : base()
+        {
+            ItemsSource = itemsSource;
+        }
+
+        /// <summary>
+        /// Base Constructor with itemsSource
+        /// </summary>
+        /// <param name="itemsSource">item's data source</param>
+        /// <param name="layouter">item's layout manager</param>
+        /// <param name="template">item's view template with data bindings</param>
+       [EditorBrowsable(EditorBrowsableState.Never)]
+        public ItemsView(IEnumerable itemsSource, ItemsLayouter layouter, DataTemplate template) : base()
+        {
+            ItemsSource = itemsSource;
+            ItemsTemplate = template;
+            ItemsLayouter = layouter;
+        }
+
+        /// <summary>
+        /// Item's source data.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public IEnumerable ItemsSource { get; set; }
+        
+        /// <summary>
+        /// DataTemplate for items.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public DataTemplate ItemsTemplate { get; set; }
+
+        /// <summary>
+        /// Items Layouter.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ItemsLayouter ItemsLayouter { get; set; }
+
+        /// <summary>
+        /// Internal encapsulated items data source.
+        /// </summary>
+       [EditorBrowsable(EditorBrowsableState.Never)]
+        internal IItemSource InternalItemSource { get; set;}
+
+        /// <summary>
+        /// RecycleCache of ViewItem.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected List<ViewItem> RecycleCache {get; set; } = new List<ViewItem>();
```

Comment:
`CA2227: Change 'RecycleCache' to be read-only by removing the property setter.`
```suggestion
        protected List<ViewItem> RecycleCache { get; } = new List<ViewItem>();
```